### PR TITLE
gl: Fix warnings and enable treat warning as error.

### PR DESCRIPTION
**Pull request by vlj**

### CODE REVIEW

File: rpcs3/Emu/RSX/GL/GLCommonDecompiler.cpp [hunk 1]
```diff
@@ -76,4 +76,5 @@ std::string compareFunctionImpl(COMPARE f, const std::string &Op0, const std::st
 	case COMPARE::FUNCTION_SNE:
 		return "notEqual(" + Op0 + ", " + Op1 + ")";
 	}
+	throw new EXCEPTION("Unknow compare function");
```
Comment (tambry) [inline rpcs3/Emu/RSX/GL/GLCommonDecompiler.cpp]: Why not:
throw EXCEPTION(...);

The latter is used everywhere instead of throw new.

**Comment by vlj** [comment]
Removed "new" before EXCEPTION

**Comment by B1ackDaemon** [comment]
@vlj, please rebase.

**Comment by vlj** [comment]
rebased

**Pull request merged**